---
title: "Transcriptomic analysis"
output: html_notebook
editor_options: 
  chunk_output_type: console
---
### Setup  
* __library(tximport):__ This package facilitates importing transcript abundance data generated from various quantification tools into R, allowing for downstream analysis and visualization. doi:[10.12688/f1000research.7563.1](https://doi.org/10.12688/f1000research.7563.1)  
* __library(tidyverse):__ Tidyverse is a collection of R packages designed for data science. It includes packages like ggplot2 for data visualization, dplyr for data manipulation, and tidyr for data tidying, providing a cohesive framework for data analysis and visualization.  
* __library(DESeq2):__ DESeq2 is a popular package for differential gene expression analysis from RNA-seq data. It uses a negative binomial distribution model to account for variability in sequencing depth and biological variability, enabling the detection of differentially expressed genes between conditions.  
* __library(phyloseq):__ Phyloseq is used for the import, storage, analysis, and graphical display of microbiome census data, particularly from high-throughput amplicon sequencing experiments such as 16S rRNA sequencing. It allows for the integration of various data types and provides tools for exploring microbial community composition and diversity.  
* __library(vegan):__ Vegan is a package for community ecology analysis, particularly focused on multivariate analysis methods. It provides functions for diversity analysis, ordination techniques (e.g., PCA, NMDS), and environmental fitting, facilitating the exploration of community structure and its relationship with environmental variables.  
  
***  

```{r setup.packages}

# if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

# BiocManager::install("tximport")
# BiocManager::install("DESeq2")
# BiocManager::install("phyloseq")
# BiocManager::install("vegan")

library(tximport)
library(tidyverse)
library(DESeq2)
library(phyloseq)
library(vegan)
```

#### Importing RNA-seq Data  
This code chunk performs the following tasks:  
1. Sets the working directory using the `setwd()` function.  
2. Reads a tabular file named "list.txt" located in the working directory into the R environment as a data frame named "samples". This file contains sample information, including sample identifiers.  
3. Constructs a vector named "files" containing the file paths to the quantification files for each sample. The paths are generated using the `file.path()` function and appending the sample names from the "samples" data frame to construct the file paths. These quantification files are in the "quant.sf" format and located within sub-directories corresponding to each sample.  
4. Uses the `tximport()` function to import transcript-level abundance estimates from the quantification files specified in the "files" vector. The argument "type = 'salmon'" indicates that the quantification files are generated by the Salmon software. The argument "txOut=TRUE" specifies that transcript-level abundance matrices should be returned.  
5. Constructs a data frame named "df" from the transcript-level abundance matrix stored in "txi.salmon$abundance".  
6. Sets the column names of the data frame "df" to be the sample names or identifiers extracted from the "samples" data frame.  


```{r setup.dataImport}
# Set the working directory to the directory containing the salmon quantification files
setwd("C:/Users/user1/Documents/R/Nitzschia/salmon_quant/")

# Read the sample information from the list.txt file into a data frame named 'samples'
samples <- read.table("C:/Users/user1/Documents/R/Nitzschia/salmon_quant/list.txt", header = TRUE)

# Construct the file paths to the quantification files for each sample
files <- file.path("C:/Users/user1/Documents/R/Nitzschia/salmon_quant/", samples$Sample, "quant.sf")

# Import transcript-level abundance estimates from the quantification files using tximport
txi.salmon <- tximport(files = files, type = "salmon", txOut=TRUE)

# Convert the transcript abundance matrix into a data frame named 'df'
df <- data.frame(txi.salmon$abundance)

# Ensure the column names are correct
print(samples$Sample)

# Set the column names of df - not working!
colnames(df) <- samples$Sample

```
  
***  
#### Pre-processing of KEGG Taxonomy Data  
In this section, we preprocess the KEGG taxonomy data obtained from diamond searches against the Nitzschia genome. Initially, we set the working directory to the folder containing the data files. Then, we read the KEGG taxonomy data from the file 'Nitzschia_keggdiamond.txt' into a data frame named 'kegg.tax'. Subsequently, we perform various data transformations, including separating columns, selecting relevant information, setting row names, duplicating columns, and converting the data frame to a matrix. These pre-processing steps are essential for subsequent analysis and visualization of the taxonomic information.

```{r setup.KEGG}
# Set the working directory to the folder containing the data files
setwd("C:/Users/user1/Documents/R/Nitzschia/")

# Read the KEGG taxonomy data from the file "Nitzschia_keggdiamond.txt" into a data frame named 'kegg.tax'
kegg.tax <- read.csv("Nitzschia_keggdiamond.txt", header = FALSE)

# Separate the first column into multiple columns based on the tab delimiter, and select only the first and third columns
kegg.tax <- kegg.tax %>% 
  separate(V1, c("R1","R2", "R3","R4","R5","R6","R7","R8","R9","R10","R11","R12"), sep = "\t") %>% 
  select(R1, R2) %>% 
  # Separate the 'R3' column into 'Species' and 'Function' based on the '|' delimiter, and select only 'R1' and 'Function'
  separate(R2, c("Species", "Function"), sep = "\\|") %>% 
  select(R1, Function)

# Set the row names of 'kegg.tax' to the values in the 'R1' column
rownames(kegg.tax) <- kegg.tax$R1

# Duplicate the 'Function' column and assign it to a new column named 'Function.1'
kegg.tax$Function.1 <- kegg.tax$Function 

# Remove the 'R1' column from 'kegg.tax'
kegg.tax <- kegg.tax %>% select(-R1)

# Convert 'kegg.tax' to a matrix
kegg.tax <- as.matrix(kegg.tax)

```

#### Colour Schemes  

```{r defineCols}
#Defining treatment colors
palette = c("Control" = "#7AC4D3", "Low" = "#E8B5D4", "Medium" = "#968CC3")

```
  
### Data Analysis
#### PCoA  
__Principal Coordinates Analysis (PCoA) and PERMANOVA__  

In this section, we conduct a Principal Coordinates Analysis (PCoA) on the Nitzschia microbiome data to explore the variation in microbial community composition across different treatment groups. We begin by reading metadata from the "Metadata.csv" file and setting up the necessary data structures. The PCoA is performed using the Bray-Curtis dissimilarity metric, and the resulting ordination plot is colored by treatment group. Additionally, we conduct PERMANOVA to assess the significance of treatment effects on community composition.  

```{r analysis.PCoA}
# Set the working directory to the folder containing the metadata file
setwd("C:/Users/user1/Documents/R/Nitzschia")

# Read the metadata from the file "Metadata.csv" into a data frame named 'metadata'
metadata <- read.csv("Metadata.csv")

# Set the row names of 'metadata' to the values in the 'SampleID' column
rownames(metadata) <- metadata$SampleID

# Convert the data frame 'df' into a matrix named 'df.mat'
df.mat <- as.matrix(df)

# Create a phyloseq object 'ps' using the OTU table, sample data, and taxonomy table
ps <- phyloseq(
  otu_table(df.mat, taxa_are_rows=TRUE),
  sample_data(metadata),
  tax_table(kegg.tax)
)

# Calculate Bray-Curtis dissimilarity between samples
ps_bray <- phyloseq::distance(ps, method="bray")

# Perform Principal Coordinates Analysis (PCoA) using the Bray-Curtis dissimilarity
ps.ord <- ordinate(ps, method="PCoA", distance=ps_bray)

# Update 'Type' variable in sample data to factor with specified levels and colors
sample_data(ps)$Type <- factor(sample_data(ps)$Type, levels=c("Control", "Low", "Medium"))

# Plot the PCoA with treatment group colors
nitzschia_plot = plot_ordination(ps, ps.ord, color = "Treatment")
nitzschia_plot_final = nitzschia_plot + geom_point(size = 2) +
  scale_color_manual(values=palette) +
  theme_classic()

# Save the plot as a PDF file and display the final plot
ggsave("Nitzschia_PCoA.pdf", width=4, height=4, units = "in", bg = "transparent")
nitzschia_plot_final
dev.off()

# Extract sample data from the phyloseq object
sd <- data.frame(sample_data(ps))

# Calculate PERMANOVA (adonis) on Bray-Curtis dissimilarity with treatment as a factor
bray.adonis <- adonis2(ps_bray ~ Treatment, data=sd, permutations=999)

# Display the results of PERMANOVA
bray.adonis

# Optional: Perform pairwise PERMANOVA
#pairwise.adonis(ps_bray, sd$Treatment, p.adjust.m='BH')
```
  
  ***  


#### Fatty acid boxplots  
This section focuses on functional analysis of the microbiome data using KEGG Orthologs (KO). The sample counts are transformed into percentages, and taxa are grouped at the functional level. Then, the functional data is enriched with KEGG pathway information obtained from "ko.kegg.csv". Finally, subsets of the data corresponding to specific KEGG pathways, such as Fatty acid biosynthesis and Fatty acid degradation, are extracted for further analysis.  

```{r analysis.FAboxPlot}
# Transform the sample counts to percentages
ps.perc = transform_sample_counts(ps, function(OTU) OTU/sum(OTU)*100)

# Group taxa at the functional level (KEGG Orthologs)
ps.perc.KO <- tax_glom(ps.perc, "Function")

# Convert the phyloseq object to a data frame
ps.perc.KO.df <- psmelt(ps.perc.KO)

# Read the KEGG pathway data
kegg.pathway <- read.csv("ko.kegg.csv", header=TRUE)

# Join the functional data with KEGG pathway information
ps.perc.KO.df.pathway <- left_join(ps.perc.KO.df, kegg.pathway, by = c("Function" = "KO"))

# Filter the data for specific KEGG pathways (e.g., Fatty acid biosynthesis)
ps.perc.KO.df.pathway.FA.biosyn <- ps.perc.KO.df.pathway %>% filter(Rank3 == "Fatty acid biosynthesis")

# Filter the data for specific KEGG pathways (e.g., Fatty acid degradation)
ps.perc.KO.df.pathway.FA.degrad <- ps.perc.KO.df.pathway %>% filter(Rank3 == "Fatty acid degradation")

```
  
  ***  
  
__Functional Analysis - Fatty Acid Degradation__  
This section focuses on the analysis of fatty acid degradation pathways within the microbiome data. The data is filtered to include only relevant KEGG functions related to fatty acid degradation. A boxplot is then created to visualize the relative abundance of these functions across different treatment groups. The plot provides insights into how the abundance of fatty acid degradation pathways varies under different experimental conditions.  

```{r analysis.functional_FAdeg}
# Assign 'Temp' values based on 'Treatment' values
ps.perc.KO.df.pathway.FA.degrad$Temp[ps.perc.KO.df.pathway.FA.degrad$Treatment == "SS"] <- "SS"
ps.perc.KO.df.pathway.FA.degrad$Temp[ps.perc.KO.df.pathway.FA.degrad$Treatment == "IS"] <- "IS"
ps.perc.KO.df.pathway.FA.degrad$Temp[ps.perc.KO.df.pathway.FA.degrad$Treatment == "SS-D"] <- "SS"

# Filter the data for specific KEGG functions related to Fatty Acid Degradation
ps.perc.KO.df.pathway.FA.degrad <- ps.perc.KO.df.pathway.FA.degrad %>% 
  filter(Function %in% c("K14338", "K00121", "K00626", "K01897", "K15013", "K00001", "K00252", "K00529", "K07511"))

# Set the levels of 'Treatment' factor
ps.perc.KO.df.pathway.FA.degrad$Treatment <- factor(ps.perc.KO.df.pathway.FA.degrad$Treatment, levels= c("SS", "SS-D", "SS-R", "IS", "IS-D", "IS-R"))

# Create a boxplot for relative abundance of KEGG functions related to Fatty Acid Degradation
p1 <- ggplot(ps.perc.KO.df.pathway.FA.degrad, aes(x=Treatment, y=Abundance, colour=Temp)) + 
  geom_boxplot() + 
  theme_classic() + 
  facet_wrap(~Function, scales = "free") +
  theme(axis.text.x = element_text(angle = 90)) +
  theme(legend.position = "") +
  ylab("Relative Abundance")

# Save the plot as a PDF file and display the final plot
ggsave("fattyacids.degrad.boxplot.pdf", width=8, height=8, units = "in", bg = "transparent")
p1
dev.off()
```
  
  __Functional Analysis - Fatty Acid Biosynthesis__  
This section focuses on the analysis of fatty acid biosynthesis pathways within the microbiome data. The data is filtered to include only relevant KEGG functions related to fatty acid biosynthesis. A boxplot is then created to visualize the relative abundance of these functions across different treatment groups. The plot provides insights into how the abundance of fatty acid biosynthesis pathways varies under different experimental conditions.  
 
```{r analysis.functional_FAbiosynthesis}
# Assign 'Temp' values based on 'Treatment' values
ps.perc.KO.df.pathway.FA.biosyn$Temp[ps.perc.KO.df.pathway.FA.biosyn$Treatment == "Control"] <- "Control"
ps.perc.KO.df.pathway.FA.biosyn$Temp[ps.perc.KO.df.pathway.FA.biosyn$Treatment == "Medium"] <- "Medium"
ps.perc.KO.df.pathway.FA.biosyn$Temp[ps.perc.KO.df.pathway.FA.biosyn$Treatment == "Low"] <- "Low"

# Filter the data for specific KEGG functions related to Fatty Acid Biosynthesis
ps.perc.KO.df.pathway.FA.biosyn <- ps.perc.KO.df.pathway.FA.biosyn %>% 
  filter(Function == "K15013")

# Set the levels of 'Treatment' factor
ps.perc.KO.df.pathway.FA.biosyn$Treatment <- factor(ps.perc.KO.df.pathway.FA.biosyn$Treatment, 
                                                    levels= c("SS", "SS-D", "SS-R", "IS", "IS-D", "IS-R"))

# Create a boxplot for relative abundance of KEGG functions related to Fatty Acid Biosynthesis
p1 <- ggplot(ps.perc.KO.df.pathway.FA.biosyn, aes(x=Treatment, y=Abundance, colour=Temp)) + 
  geom_boxplot() + 
  theme_classic() + 
  facet_wrap(~Function, scales = "free") +
  theme(axis.text.x = element_text(angle = 90)) +
  theme(legend.position = "") +
  ylab("Relative Abundance")

# Save the plot as a PDF file and display the final plot
ggsave("fattyacids.biosyn.boxplot.pdf", width=4, height=4, units = "in", bg = "transparent")
p1
dev.off()

```
  
  ***  
  

#### Differential Abundance Analysis  
This section performs a differential abundance analysis using DESeq2 to identify KEGG functions that are significantly differentially abundant between treatment groups (SS and SS-D). The results are visualized as a scatter plot showing log2 fold changes for each function. The analysis provides insights into functional changes in the microbiome associated with different treatments.  

```{r analysis.DAA}
# Group taxa at the functional level (KEGG Orthologs)
ps.merged <- tax_glom(ps, "Function")

# Convert phyloseq object to DESeq2 object
dds <- phyloseq_to_deseq2(ps.merged, ~ Treatment)

# Perform DESeq2 analysis
set.seed(66)
dds2 = DESeq(dds, test="Wald", fitType="parametric", sfType="iterate")

# Load necessary libraries and set theme for ggplot
library("ggplot2")
theme_set(theme_bw())

# Function to set discrete fill scale
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}

# Extract results for specified contrasts and adjust p-values
resSS.SSD = results(dds2, contrast = c("Treatment", "SS-D", "SS"), cooksCutoff = FALSE, pAdjustMethod = "BH", independentFiltering=TRUE)

# Filter significant results based on adjusted p-value
alpha = 0.05
sigtabSS.SSD = resSS.SSD[which(resSS.SSD$padj < alpha), ]

# Add taxonomic information to the results
resSS.SSD = cbind(as(resSS.SSD, "data.frame"), as(tax_table(ps)[rownames(resSS.SSD), ], "matrix"))
resSS.SSD["Comparison"]<- "SS - SS-D"
sigtabSS.SSD = cbind(as(sigtabSS.SSD, "data.frame"), as(tax_table(ps)[rownames(sigtabSS.SSD), ], "matrix"))
sigtabSS.SSD["Comparison"]<- "SS - SS-D"

# Order the functions based on maximum absolute log2 fold change
x = tapply(sigtabSS.SSD$log2FoldChange, sigtabSS.SSD$Function, function(x) max(x))
x = sort(x, TRUE)
sigtabSS.SSD$Function = factor(as.character(sigtabSS.SSD$Function), levels=names(x))

# Create a scatter plot of log2 fold changes for significant results
SS.SSD.plot <- ggplot(sigtabSS.SSD, aes(x=Function, y=log2FoldChange)) + geom_point(size=3) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))

```
  
  ***  
  


```{r}
resSS.SSR = results(dds2, contrast = c("Treatment", "SS", "SS-R"), cooksCutoff = FALSE, pAdjustMethod = "BH", independentFiltering=TRUE)
alpha = 0.05
sigtabSS.SSR = resSS.SSR[which(resSS.SSR$padj < alpha), ]
#sigtabNC = sigtabNC[which(sigtabNC$log2FoldChange > 2 | sigtabNC$log2FoldChange < -2), ]
resSS.SSR = cbind(as(resSS.SSR, "data.frame"), as(tax_table(ps)[rownames(resSS.SSR), ], "matrix"))
resSS.SSR["Comparison"]<- "SS - SS-R"
sigtabSS.SSR = cbind(as(sigtabSS.SSR, "data.frame"), as(tax_table(ps)[rownames(sigtabSS.SSR), ], "matrix"))
sigtabSS.SSR["Comparison"]<- "SS - SS-R"


x = tapply(sigtabSS.SSR$log2FoldChange, sigtabSS.SSR$Function, function(x) max(x))
x = sort(x, TRUE)
sigtabSS.SSR$Function = factor(as.character(sigtabSS.SSR$Function), levels=names(x))

SS.SSR.plot <- ggplot(sigtabSS.SSR, aes(x=Function, y=log2FoldChange)) + geom_point(size=3) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
```



```{r}
resSSD.SSR = results(dds2, contrast = c("Treatment", "SS-D", "SS-R"), cooksCutoff = FALSE, pAdjustMethod = "BH", independentFiltering=TRUE)
alpha = 0.05
sigtabSSD.SSR = resSSD.SSR[which(resSSD.SSR$padj < alpha), ]
#sigtabNC = sigtabNC[which(sigtabNC$log2FoldChange > 2 | sigtabNC$log2FoldChange < -2), ]
resSSD.SSR = cbind(as(resSSD.SSR, "data.frame"), as(tax_table(ps)[rownames(resSSD.SSR), ], "matrix"))
resSSD.SSR["Comparison"]<- "SS-D - SS-R"
sigtabSSD.SSR = cbind(as(sigtabSSD.SSR, "data.frame"), as(tax_table(ps)[rownames(sigtabSSD.SSR), ], "matrix"))
sigtabSSD.SSR["Comparison"]<- "SS-D - SS-R"


x = tapply(sigtabSSD.SSR$log2FoldChange, sigtabSSD.SSR$Function, function(x) max(x))
x = sort(x, TRUE)
sigtabSSD.SSR$Function = factor(as.character(sigtabSSD.SSR$Function), levels=names(x))

SSD.SSR.plot <- ggplot(sigtabSSD.SSR, aes(x=Function, y=log2FoldChange)) + geom_point(size=3) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
```


```{r}
resIS.ISD = results(dds2, contrast = c("Treatment", "IS", "IS-D"), cooksCutoff = FALSE, pAdjustMethod = "BH", independentFiltering=TRUE)
alpha = 0.05
sigtabIS.ISD = resIS.ISD[which(resIS.ISD$padj < alpha), ]
#sigtabNC = sigtabNC[which(sigtabNC$log2FoldChange > 2 | sigtabNC$log2FoldChange < -2), ]
resIS.ISD = cbind(as(resIS.ISD, "data.frame"), as(tax_table(ps)[rownames(resIS.ISD), ], "matrix"))
resIS.ISD["Comparison"]<- "IS - IS-D"
sigtabIS.ISD = cbind(as(sigtabIS.ISD, "data.frame"), as(tax_table(ps)[rownames(sigtabIS.ISD), ], "matrix"))
sigtabIS.ISD["Comparison"]<- "IS - IS-D"


x = tapply(sigtabIS.ISD$log2FoldChange, sigtabIS.ISD$Function, function(x) max(x))
x = sort(x, TRUE)
sigtabIS.ISD$Function = factor(as.character(sigtabIS.ISD$Function), levels=names(x))

IS.ISD.plot <- ggplot(sigtabIS.ISD, aes(x=Function, y=log2FoldChange)) + geom_point(size=3) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
```


```{r}
resIS.ISR = results(dds2, contrast = c("Treatment", "IS", "IS-R"), cooksCutoff = FALSE, pAdjustMethod = "BH", independentFiltering=TRUE)
alpha = 0.05
sigtabIS.ISR = resIS.ISR[which(resIS.ISR$padj < alpha), ]
#sigtabNC = sigtabNC[which(sigtabNC$log2FoldChange > 2 | sigtabNC$log2FoldChange < -2), ]
resIS.ISR = cbind(as(resIS.ISR, "data.frame"), as(tax_table(ps)[rownames(resIS.ISR), ], "matrix"))
resIS.ISR["Comparison"]<- "IS - IS-R"
sigtabIS.ISR = cbind(as(sigtabIS.ISR, "data.frame"), as(tax_table(ps)[rownames(sigtabIS.ISR), ], "matrix"))
sigtabIS.ISR["Comparison"]<- "IS - IS-R"


x = tapply(sigtabIS.ISR$log2FoldChange, sigtabIS.ISR$Function, function(x) max(x))
x = sort(x, TRUE)
sigtabIS.ISR$Function = factor(as.character(sigtabIS.ISR$Function), levels=names(x))

IS.ISR.plot <- ggplot(sigtabIS.ISR, aes(x=Function, y=log2FoldChange)) + geom_point(size=3) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
```



```{r}
resISD.ISR = results(dds2, contrast = c("Treatment", "IS-D", "IS-R"), cooksCutoff = FALSE, pAdjustMethod = "BH", independentFiltering=TRUE)
alpha = 0.05
sigtabISD.ISR = resISD.ISR[which(resISD.ISR$padj < alpha), ]
#sigtabNC = sigtabNC[which(sigtabNC$log2FoldChange > 2 | sigtabNC$log2FoldChange < -2), ]
resISD.ISR = cbind(as(resISD.ISR, "data.frame"), as(tax_table(ps)[rownames(resISD.ISR), ], "matrix"))
resISD.ISR["Comparison"]<- "IS-D - IS-R"
sigtabISD.ISR = cbind(as(sigtabISD.ISR, "data.frame"), as(tax_table(ps)[rownames(sigtabISD.ISR), ], "matrix"))
sigtabISD.ISR["Comparison"]<- "IS-D - IS-R"


x = tapply(sigtabISD.ISR$log2FoldChange, sigtabISD.ISR$Function, function(x) max(x))
x = sort(x, TRUE)
sigtabISD.ISR$Function = factor(as.character(sigtabISD.ISR$Function), levels=names(x))

ISD.ISR.plot <- ggplot(sigtabISD.ISR, aes(x=Function, y=log2FoldChange)) + geom_point(size=3) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
```





```{r}
resSS.IS = results(dds2, contrast = c("Treatment", "SS", "IS"), cooksCutoff = FALSE, pAdjustMethod = "BH", independentFiltering=TRUE)
alpha = 0.05
sigtabSS.IS = resSS.IS[which(resSS.IS$padj < alpha), ]
#sigtabNC = sigtabNC[which(sigtabNC$log2FoldChange > 2 | sigtabNC$log2FoldChange < -2), ]
resSS.IS = cbind(as(resSS.IS, "data.frame"), as(tax_table(ps)[rownames(resSS.IS), ], "matrix"))
resSS.IS["Comparison"]<- "SS - IS"
sigtabSS.IS = cbind(as(sigtabSS.IS, "data.frame"), as(tax_table(ps)[rownames(sigtabSS.IS), ], "matrix"))
sigtabSS.IS["Comparison"]<- "SS - IS"


x = tapply(sigtabSS.IS$log2FoldChange, sigtabSS.IS$Function, function(x) max(x))
x = sort(x, TRUE)
sigtabSS.IS$Function = factor(as.character(sigtabSS.IS$Function), levels=names(x))

SS.IS.plot <- ggplot(sigtabSS.IS, aes(x=Function, y=log2FoldChange)) + geom_point(size=3) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
```



```{r}
resSSD.ISD = results(dds2, contrast = c("Treatment", "SS-D", "IS-D"), cooksCutoff = FALSE, pAdjustMethod = "BH", independentFiltering=TRUE)
alpha = 0.05
sigtabSSD.ISD = resSSD.ISD[which(resSSD.ISD$padj < alpha), ]
#sigtabNC = sigtabNC[which(sigtabNC$log2FoldChange > 2 | sigtabNC$log2FoldChange < -2), ]
resSSD.ISD = cbind(as(resSSD.ISD, "data.frame"), as(tax_table(ps)[rownames(resSSD.ISD), ], "matrix"))
resSSD.ISD["Comparison"]<- "SS-D - IS-D"
sigtabSSD.ISD = cbind(as(sigtabSSD.ISD, "data.frame"), as(tax_table(ps)[rownames(sigtabSSD.ISD), ], "matrix"))
sigtabSSD.ISD["Comparison"]<- "SS-D - IS-D"


x = tapply(sigtabSSD.ISD$log2FoldChange, sigtabSSD.ISD$Function, function(x) max(x))
x = sort(x, TRUE)
sigtabSSD.ISD$Function = factor(as.character(sigtabSSD.ISD$Function), levels=names(x))

SSD.ISD.plot <- ggplot(sigtabSSD.ISD, aes(x=Function, y=log2FoldChange)) + geom_point(size=3) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
```



```{r}
resSSR.ISR = results(dds2, contrast = c("Treatment", "SS-R", "IS-R"), cooksCutoff = FALSE, pAdjustMethod = "BH", independentFiltering=TRUE)
alpha = 0.05
sigtabSSR.ISR = resSSR.ISR[which(resSSR.ISR$padj < alpha), ]
#sigtabNC = sigtabNC[which(sigtabNC$log2FoldChange > 2 | sigtabNC$log2FoldChange < -2), ]
resSSR.ISR = cbind(as(resSSR.ISR, "data.frame"), as(tax_table(ps)[rownames(resSSR.ISR), ], "matrix"))
resSSR.ISR["Comparison"]<- "SS-R - IS-R"
sigtabSSR.ISR = cbind(as(sigtabSSR.ISR, "data.frame"), as(tax_table(ps)[rownames(sigtabSSR.ISR), ], "matrix"))
sigtabSSR.ISR["Comparison"]<- "SS-R - IS-R"


x = tapply(sigtabSSR.ISR$log2FoldChange, sigtabSSR.ISR$Function, function(x) max(x))
x = sort(x, TRUE)
sigtabSSR.ISR$Function = factor(as.character(sigtabSSR.ISR$Function), levels=names(x))

SSR.ISR.plot <- ggplot(sigtabSSR.ISR, aes(x=Function, y=log2FoldChange)) + geom_point(size=3) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
```



```{r}
combined <- rbind(sigtabSS.SSD, sigtabSS.SSR, sigtabSSD.SSR, sigtabIS.ISD, sigtabIS.ISR, sigtabISD.ISR,
                  sigtabSS.IS, sigtabSSD.ISD, sigtabSSR.ISR)


kegg.pathway <- read.csv("ko.kegg.csv", h=T)

combined <- left_join(combined, kegg.pathway, by = c("Function.1" = "KO"))


write.csv(combined, "all.differentially.expressed.csv", row.names = F)
```





```{r}


combined$Type[combined$log2FoldChange < 0] <- "Negative"
combined$Type[combined$log2FoldChange > 0] <- "Positive"

combined$Rank2[combined$Rank2 == "Protein families: metabolism"] <- "Other Metabolism"
combined$Rank2[combined$Rank2 == "Protein families: genetic information processing"]   <- "Genetic Information Processing"
combined$Rank2[combined$Rank2 == "Protein families: signaling and cellular processes"] <- "Environmental Information Processing"
combined$Rank2[combined$Rank2 == "Unclassified: metabolism"] <- "Other Metabolism"
combined$Rank2[combined$Rank2 == "Unclassified: genetic information processing"] <- "Other Genetic Information Processing"
combined$Rank2[combined$Rank2 == "Unclassified: signaling and cellular processes"] <- "Environmental Information Processing"
combined$Rank2[combined$Rank2 == "Protein families: signaling and cellular processes"] <- "Other Signaling and cellular processes"




combined1 <- combined %>%
  filter(Comparison != "SS - IS")  %>% filter(Comparison != "SS-D - IS-D")  %>% filter(Comparison != "SS-R - IS-R") %>% drop_na() %>%
  select(-Rank3, -Rank4) %>%
  group_by(Comparison, Rank2) %>% distinct()

combined1 <- data.frame(combined1)

summary1 <- combined1 %>%
  group_by(Comparison, Rank2) %>%
  tally() %>%
  filter(n >= 5)


summary <- combined1 %>%
  group_by(Comparison, Type) %>% 
  tally()

summary2 <- combined1 %>%
  filter(Rank2 == "Amino acid metabolism" | 
         #Rank2 == "Biosynthesis of other secondary metabolites" |
         Rank2 == "Carbohydrate metabolism" |   
         #Rank2 == "Cell growth and death" |  
         Rank2 == "Energy metabolism" |  
         Rank2 == "Environmental Information Processing" |  
         Rank2 == "Folding, sorting and degradation" |  
         Rank2 == "Genetic Information Processing" |  
         #Rank2 == "Lipid metabolism" |  
         Rank2 == "Metabolism of cofactors and vitamins" |  
         Rank2 == "Nucleotide metabolism" |  
         Rank2 == "Other Metabolism" |  
         Rank2 == "Signal transduction" |
         Rank2 == "Translation" |
         Rank2 == "Transport and catabolism" |
         Rank2 == "Xenobiotics biodegradation and metabolism") %>%
  group_by(Comparison, Type) %>%
  tally()




summary3 <- combined1 %>%
  filter(Rank2 == "Amino acid metabolism" | 
         #Rank2 == "Biosynthesis of other secondary metabolites" |
         Rank2 == "Carbohydrate metabolism" |   
         #Rank2 == "Cell growth and death" |  
         Rank2 == "Energy metabolism" |  
         Rank2 == "Environmental Information Processing" |  
         Rank2 == "Folding, sorting and degradation" |  
         Rank2 == "Genetic Information Processing" |  
         #Rank2 == "Lipid metabolism" |  
         Rank2 == "Metabolism of cofactors and vitamins" |  
         Rank2 == "Nucleotide metabolism" |  
         Rank2 == "Other Metabolism" |  
         Rank2 == "Signal transduction" |
         Rank2 == "Translation" |
         Rank2 == "Transport and catabolism" |
         Rank2 == "Xenobiotics biodegradation and metabolism") %>%
  group_by(Comparison, Rank2, Type) %>%
  tally() 

summary3 <- data.frame(summary3)
summary4 <- summary3 %>%
  tibble::add_row(Comparison = "SS - SS-D", Rank2 = "Other", Type = "Positive", n = 13) %>%
  tibble::add_row(Comparison = "SS - SS-R", Rank2 = "Other", Type = "Positive", n = 1) %>%
  tibble::add_row(Comparison = "IS - IS-D", Rank2 = "Other", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "IS - IS-R", Rank2 = "Other", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "SS-D - SS-R", Rank2 = "Other", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Other", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "SS - SS-D", Rank2 = "Other", Type = "Negative", n = 15) %>%
  tibble::add_row(Comparison = "SS - SS-R", Rank2 = "Other", Type = "Negative", n = 13) %>%
  tibble::add_row(Comparison = "IS - IS-D", Rank2 = "Other", Type = "Negative", n = 12) %>%
  tibble::add_row(Comparison = "IS - IS-R", Rank2 = "Other", Type = "Negative", n = 12) %>%
  tibble::add_row(Comparison = "SS-D - SS-R", Rank2 = "Other", Type = "Negative", n = 2) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Other", Type = "Negative", n = 4)
```

```{r}

summary4.SS.SSD <- summary4 %>% filter(Comparison == "SS - SS-D")

summary4.SS.SSD$Rank2 <- factor(summary4.SS.SSD$Rank2, levels = c("Other", "Genetic Information Processing", "Environmental Information Processing", "Signal transduction", 
                                                          "Folding, sorting and degradation", "Transport and catabolism", "Translation",
                                                          "Other Metabolism", "Xenobiotics biodegradation and metabolism", "Metabolism of cofactors and vitamins",
                                                          "Nucleotide metabolism", "Energy metabolism", "Carbohydrate metabolism", "Amino acid metabolism"))


summary4.SS.SSD$Type[summary4.SS.SSD$Type == "Negative"] <- "SS"
summary4.SS.SSD$Type[summary4.SS.SSD$Type == "Positive"] <- "SS-D"


plot.SS.SSD <- ggplot(summary4.SS.SSD, aes(Rank2)) + 
  geom_bar(data = subset(summary4.SS.SSD, Type == "SS-D"), aes(y = n, fill = Type), stat = "identity", position = "dodge") +
  geom_bar(data = subset(summary4.SS.SSD, Type == "SS"), aes(y = -n, fill = Type), stat = "identity", position = "dodge") + 
  geom_hline(yintercept = 0,colour = "grey90")  + 
  theme_bw() +
  xlab("KEGG Pathway") +
  ylab("") +
  scale_fill_manual(values = palette) +
  scale_y_continuous(limits = c(-100, 100), breaks = seq(-100,100,by = 25))  + 
  theme(axis.text.y = element_text(size = 14),
        axis.text.x = element_blank(), 
        axis.ticks.y = element_blank(),
        axis.title.x = element_blank(),
        axis.title = element_text(size=14),
        panel.grid = element_blank()) +
  theme(legend.position = "") +
  geom_text(data = subset(summary4.SS.SSD, Type == "SS-D"), aes(Rank2, n, label=n), position = position_dodge(width=0.9), hjust = -0.5, size=4) +
  geom_text(data = subset(summary4.SS.SSD, Type == "SS"), aes(Rank2, -n, label=n),  position = position_dodge(width=0.9), hjust = 1.5, size=4) +
  coord_flip()

plot.final.SS.SSD <- plot.SS.SSD + annotate("text", x = 14, y = -70, label = "SS", size = 5) + annotate("text", x = 14, y = 70, label = "SS-D", size = 5)
```






```{r}

summary4.SS.SSR <- summary4 %>% filter(Comparison == "SS - SS-R")

summary4.SS.SSR$Rank2 <- factor(summary4.SS.SSR$Rank2, levels = c("Other", "Genetic Information Processing", "Environmental Information Processing", "Signal transduction", 
                                                          "Folding, sorting and degradation", "Transport and catabolism", "Translation",
                                                          "Other Metabolism", "Xenobiotics biodegradation and metabolism", "Metabolism of cofactors and vitamins",
                                                          "Nucleotide metabolism", "Energy metabolism", "Carbohydrate metabolism", "Amino acid metabolism"))


summary4.SS.SSR$Type[summary4.SS.SSR$Type == "Negative"] <- "SS-R"
summary4.SS.SSR$Type[summary4.SS.SSR$Type == "Positive"] <- "SS"


plot.SS.SSR <- ggplot(summary4.SS.SSR, aes(Rank2)) + 
  geom_bar(data = subset(summary4.SS.SSR, Type == "SS-R"), aes(y = n, fill = Type), stat = "identity", position = "dodge") +
  geom_bar(data = subset(summary4.SS.SSR, Type == "SS"), aes(y = -n, fill = Type), stat = "identity", position = "dodge") + 
  geom_hline(yintercept = 0,colour = "grey90")  + 
  theme_bw() +
  xlab("KEGG Pathway") +
  ylab("") +
  scale_fill_manual(values = palette) +
  scale_y_continuous(limits = c(-100, 100), breaks = seq(-100,100,by = 25))  + 
  theme(axis.text = element_blank(), 
        axis.title = element_blank(),
        panel.grid = element_blank()) +
  theme(legend.position = "") +
  geom_text(data = subset(summary4.SS.SSR, Type == "SS-R"), aes(Rank2, n, label=n), position = position_dodge(width=0.9), hjust = -0.5, size=4) +
  geom_text(data = subset(summary4.SS.SSR, Type == "SS"), aes(Rank2, -n, label=n),  position = position_dodge(width=0.9), hjust = 1.5, size=4) +
  coord_flip()

plot.final.SS.SSR <- plot.SS.SSR + annotate("text", x = 14, y = -70, label = "SS", size = 5) + annotate("text", x = 14, y = 70, label = "SS-R", size = 5)
```





```{r}

summary4.SSD.SSR <- summary4 %>% filter(Comparison == "SS-D - SS-R")

summary4.SSD.SSR <- summary4.SSD.SSR %>% 
  tibble::add_row(Comparison = "SS-D - SS-R", Rank2 = "Folding, sorting and degradation", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "SS-D - SS-R", Rank2 = "Transport and catabolism", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "SS-D - SS-R", Rank2 = "Energy metabolism", Type = "Negative", n = NA) %>%
  tibble::add_row(Comparison = "SS-D - SS-R", Rank2 = "Genetic Information Processing", Type = "Negative", n = NA) %>%
  tibble::add_row(Comparison = "SS-D - SS-R", Rank2 = "Environmental Information Processing", Type = "Negative", n = NA) %>%
  tibble::add_row(Comparison = "SS-D - SS-R", Rank2 = "Metabolism of cofactors and vitamins", Type = "Negative", n = NA)%>%
  tibble::add_row(Comparison = "SS-D - SS-R", Rank2 = "Nucleotide metabolism", Type = "Negative", n = NA)%>%
  tibble::add_row(Comparison = "SS-D - SS-R", Rank2 = "Translation", Type = "Negative", n = NA)%>%
  tibble::add_row(Comparison = "SS-D - SS-R", Rank2 = "Xenobiotics biodegradation and metabolism", Type = "Negative", n = NA)


summary4.SSD.SSR <- as.data.frame(as.matrix(summary4.SSD.SSR))

summary4.SSD.SSR$Rank2 <- factor(summary4.SSD.SSR$Rank2, levels = c("Other", "Genetic Information Processing", "Environmental Information Processing", "Signal transduction", 
                                                          "Folding, sorting and degradation", "Transport and catabolism", "Translation",
                                                          "Other Metabolism", "Xenobiotics biodegradation and metabolism", "Metabolism of cofactors and vitamins",
                                                          "Nucleotide metabolism", "Energy metabolism", "Carbohydrate metabolism", "Amino acid metabolism"))


summary4.SSD.SSR$Type[summary4.SSD.SSR$Type == "Negative"] <- "SS-R"
summary4.SSD.SSR$Type[summary4.SSD.SSR$Type == "Positive"] <- "SS-D"

summary4.SSD.SSR$n <- as.numeric(summary4.SSD.SSR$n)


plot.SSD.SSR <- ggplot(summary4.SSD.SSR, aes(Rank2)) + 
  geom_bar(data = subset(summary4.SSD.SSR, Type == "SS-D"), aes(y = -n, fill = Type), stat = "identity", position = "dodge") +
  geom_bar(data = subset(summary4.SSD.SSR, Type == "SS-R"), aes(y = n, fill = Type), stat = "identity", position = "dodge") +
  geom_hline(yintercept = 0,colour = "grey90")  + 
  theme_bw() +
  xlab("KEGG Pathway") +
  ylab("") +
  scale_fill_manual(values = palette) +
  scale_y_continuous(limits = c(-100, 100), breaks = seq(-100,100,by = 25))  + 
  theme(axis.text = element_blank(), 
        axis.title = element_blank(),
        panel.grid = element_blank()) +
  theme(legend.position = "") +
  geom_text(data = subset(summary4.SSD.SSR, Type == "SS-R"), aes(Rank2, n, label=n), position = position_dodge(width=0.9), hjust = -0.5, size=4) +
  geom_text(data = subset(summary4.SSD.SSR, Type == "SS-D"), aes(Rank2, -n, label=n),  position = position_dodge(width=0.9), hjust = 1.5, size=4) +
  coord_flip() 

plot.final.SSD.SSR <- plot.SSD.SSR + annotate("text", x = 14, y = -70, label = "SS-D", size = 5) + annotate("text", x = 14, y = 70, label = "SS-R", size = 5)
```


```{r}

summary4.IS.ISD <- summary4 %>% filter(Comparison == "IS - IS-D")

summary4.IS.ISD$Rank2 <- factor(summary4.IS.ISD$Rank2, levels = c("Other", "Genetic Information Processing", "Environmental Information Processing", "Signal transduction", 
                                                          "Folding, sorting and degradation", "Transport and catabolism", "Translation",
                                                          "Other Metabolism", "Xenobiotics biodegradation and metabolism", "Metabolism of cofactors and vitamins",
                                                          "Nucleotide metabolism", "Energy metabolism", "Carbohydrate metabolism", "Amino acid metabolism"))


summary4.IS.ISD$Type[summary4.IS.ISD$Type == "Positive"] <- "IS"
summary4.IS.ISD$Type[summary4.IS.ISD$Type == "Negative"] <- "IS-D"


plot.IS.ISD <- ggplot(summary4.IS.ISD, aes(Rank2)) + 
  geom_bar(data = subset(summary4.IS.ISD, Type == "IS-D"), aes(y = n, fill = Type), stat = "identity", position = "dodge") +
  geom_bar(data = subset(summary4.IS.ISD, Type == "IS"), aes(y = -n, fill = Type), stat = "identity", position = "dodge") + 
  geom_hline(yintercept = 0,colour = "grey90")  + 
  theme_bw() +
  xlab("KEGG Pathway") +
  ylab("") +
  scale_fill_manual(values = palette) +
  scale_y_continuous(limits = c(-100, 100), breaks = seq(-100,100,by = 25))  + 
  theme(axis.text.y = element_text(size = 14),
        axis.text.x = element_blank(), 
        axis.ticks.y = element_blank(),
        axis.title.x = element_blank(),
        axis.title = element_text(size=14),
        panel.grid = element_blank()) +
  theme(legend.position = "") +
  geom_text(data = subset(summary4.IS.ISD, Type == "IS-D"), aes(Rank2, n, label=n), position = position_dodge(width=0.9), hjust = -0.5, size=4) +
  geom_text(data = subset(summary4.IS.ISD, Type == "IS"), aes(Rank2, -n, label=n),  position = position_dodge(width=0.9), hjust = 1.5, size=4) +
  coord_flip()

plot.final.IS.ISD <- plot.IS.ISD + annotate("text", x = 14, y = -70, label = "IS", size = 5) + annotate("text", x = 14, y = 70, label = "IS-D", size = 5)
```






```{r}

summary4.IS.ISR <- summary4 %>% filter(Comparison == "IS - IS-R")

summary4.IS.ISR <- summary4.IS.ISR %>% 
  tibble::add_row(Comparison = "IS - IS-R", Rank2 = "Nucleotide metabolism", Type = "Positive", n = NA) %>% 
  tibble::add_row(Comparison = "IS - IS-R", Rank2 = "Nucleotide metabolism", Type = "Negative", n = NA)

summary4.IS.ISR <- as.data.frame(as.matrix(summary4.IS.ISR))

summary4.IS.ISR$Rank2 <- factor(summary4.IS.ISR$Rank2, levels = c("Other", "Genetic Information Processing", "Environmental Information Processing", "Signal transduction", 
                                                          "Folding, sorting and degradation", "Transport and catabolism", "Translation",
                                                          "Other Metabolism", "Xenobiotics biodegradation and metabolism", "Metabolism of cofactors and vitamins",
                                                          "Nucleotide metabolism", "Energy metabolism", "Carbohydrate metabolism", "Amino acid metabolism"))


summary4.IS.ISR$Type[summary4.IS.ISR$Type == "Negative"] <- "IS-R"
summary4.IS.ISR$Type[summary4.IS.ISR$Type == "Positive"] <- "IS"

summary4.IS.ISR$n <- as.numeric(summary4.IS.ISR$n)

plot.IS.ISR <- ggplot(summary4.IS.ISR, aes(Rank2)) + 
  geom_bar(data = subset(summary4.IS.ISR, Type == "IS-R"), aes(y = n, fill = Type), stat = "identity", position = "dodge") +
  geom_bar(data = subset(summary4.IS.ISR, Type == "IS"), aes(y = -n, fill = Type), stat = "identity", position = "dodge") +
  geom_hline(yintercept = 0,colour = "grey90")  + 
  theme_bw() +
  xlab("KEGG Pathway") +
  ylab("") +
  scale_fill_manual(values = palette) +
  scale_y_continuous(limits = c(-100, 100), breaks = seq(-100,100,by = 25))  + 
  theme(axis.text = element_blank(), 
        axis.title = element_blank(),
        panel.grid = element_blank()) +
  theme(legend.position = "") +
  geom_text(data = subset(summary4.IS.ISR, Type == "IS-R"), aes(Rank2, n, label=n), position = position_dodge(width=0.9), hjust = -0.5, size=4) +
  geom_text(data = subset(summary4.IS.ISR, Type == "IS"), aes(Rank2, -n, label=n),  position = position_dodge(width=0.9), hjust = 1.5, size=4) +
  coord_flip()

plot.final.IS.ISR <- plot.IS.ISR + annotate("text", x = 14, y = -70, label = "IS", size = 5) + annotate("text", x = 14, y = 70, label = "IS-R", size = 5)
```


```{r}

summary4.ISD.ISR <- summary4 %>% filter(Comparison == "IS-D - IS-R")

summary4.ISD.ISR <- summary4.ISD.ISR %>% 
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Amino acid metabolism", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Energy metabolism", Type = "Negative", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Folding, sorting and degradation", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Translation", Type = "Negative", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Transport and catabolism", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Carbohydrate metabolism", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Nucleotide metabolism", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Metabolism of cofactors and vitamins", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Xenobiotics biodegradation and metabolism", Type = "Positive", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Carbohydrate metabolism", Type = "Negative", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Nucleotide metabolism", Type = "Negative", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Metabolism of cofactors and vitamins", Type = "Negative", n = NA) %>%
  tibble::add_row(Comparison = "IS-D - IS-R", Rank2 = "Xenobiotics biodegradation and metabolism", Type = "Negative", n = NA) 

summary4.ISD.ISR <- as.data.frame(as.matrix(summary4.ISD.ISR))

summary4.ISD.ISR$Rank2 <- factor(summary4.ISD.ISR$Rank2, levels = c("Other", "Genetic Information Processing", "Environmental Information Processing", "Signal transduction", 
                                                          "Folding, sorting and degradation", "Transport and catabolism", "Translation",
                                                          "Other Metabolism", "Xenobiotics biodegradation and metabolism", "Metabolism of cofactors and vitamins",
                                                          "Nucleotide metabolism", "Energy metabolism", "Carbohydrate metabolism", "Amino acid metabolism"))


summary4.ISD.ISR$Type[summary4.ISD.ISR$Type == "Negative"] <- "IS-R"
summary4.ISD.ISR$Type[summary4.ISD.ISR$Type == "Positive"] <- "IS-D"
summary4.ISD.ISR$n <- as.numeric(summary4.ISD.ISR$n)


plot.ISD.ISR <- ggplot(summary4.ISD.ISR, aes(Rank2)) + 
  geom_bar(data = subset(summary4.ISD.ISR, Type == "IS-R"), aes(y = n, fill = Type), stat = "identity", position = "dodge") +
  geom_bar(data = subset(summary4.ISD.ISR, Type == "IS-D"), aes(y = -n, fill = Type), stat = "identity", position = "dodge") + 
  geom_hline(yintercept = 0,colour = "grey90")  + 
  theme_bw() +
  xlab("KEGG Pathway") +
  ylab("") +
  scale_fill_manual(values = palette) +
  scale_y_continuous(limits = c(-100, 100), breaks = seq(-100,100,by = 25))  + 
  theme(axis.text = element_blank(), 
        axis.title = element_blank(),
        panel.grid = element_blank()) +
  theme(legend.position = "") +
  geom_text(data = subset(summary4.ISD.ISR, Type == "IS-R"), aes(Rank2, n, label=n), position = position_dodge(width=0.9), hjust = -0.5, size=4) +
  geom_text(data = subset(summary4.ISD.ISR, Type == "IS-D"), aes(Rank2, -n, label=n),  position = position_dodge(width=0.9), hjust = 1.5, size=4) +
  coord_flip() 

plot.final.ISD.ISR <- plot.ISD.ISR + annotate("text", x = 14, y = -70, label = "IS-D", size = 5) + annotate("text", x = 14, y = 70, label = "IS-R", size = 5)
```




```{r}
jpeg("Differential.expressed.v1.jpeg", width=20, height=20, res=300, units = "in", bg = "transparent")
gl<-grid.layout(nrow=2, ncol=3, heights=unit(c(20), "null"), widths=unit(c(30,20,20),"null"))
vp.1<-viewport(layout.pos.col=1, layout.pos.row=1)
vp.2<-viewport(layout.pos.col=2, layout.pos.row=1)
vp.3<-viewport(layout.pos.col=3, layout.pos.row=1)
vp.4<-viewport(layout.pos.col=1, layout.pos.row=2)
vp.5<-viewport(layout.pos.col=2, layout.pos.row=2)
vp.6<-viewport(layout.pos.col=3, layout.pos.row=2)
pushViewport(viewport(layout=gl))
print(plot.final.SS.SSD + theme(legend.position = "none"), vp=vp.1)
print(plot.final.SS.SSR + theme(legend.position = "none"), vp=vp.2)
print(plot.final.SSD.SSR + theme(legend.position = "none"), vp=vp.3)

print(plot.final.IS.ISD + theme(legend.position = "none"), vp=vp.4)
print(plot.final.IS.ISR + theme(legend.position = "none"), vp=vp.5)
print(plot.final.ISD.ISR + theme(legend.position = "none"), vp=vp.6)

dev.off()
```

#### FA PCoA

```{r}

ps.perc = transform_sample_counts(ps, function(OTU) OTU/sum(OTU)*100)


ps.fatty.acids <- subset_taxa(ps, 
                                      Function == "K00022" |
Function == "K00059" |
Function == "K00208" |
Function == "K00209" |
Function == "K00645" |
Function == "K00647" |
Function == "K00648" |
Function == "K00659" |
Function == "K00665" |
Function == "K00667" |
Function == "K00668" |
Function == "K01068" |
Function == "K01071" |
Function == "K01074" |
Function == "K01716" |
Function == "K01897" |
Function == "K02371" |
Function == "K02372" |
Function == "K03921" |
Function == "K03922" |
Function == "K03955" |
Function == "K07508" |
Function == "K07509" |
Function == "K07511" |
Function == "K07512" |
Function == "K07515" |
Function == "K09458" |
Function == "K10203" |
Function == "K10205" |
Function == "K10244" |
Function == "K10245" |
Function == "K10246" |
Function == "K10247" |
Function == "K10248" |
Function == "K10249" |
Function == "K10250" |
Function == "K10251" |
Function == "K10258" |
Function == "K10703" |
Function == "K10780" |
Function == "K10781" |
Function == "K10804" |
Function == "K10805" |
Function == "K10806" |
Function == "K11262" |
Function == "K11533" |
Function == "K11539" |
Function == "K13370" |
Function == "K15013" |
Function == "K15397" |
Function == "K16363" |
Function == "K16399" |
Function == "K17360" |
Function == "K18473" |
Function == "K18474" |
Function == "K18660" |
Function == "K22540" |
Function == "K22541" |
Function == "K22554" |
Function == "K01946" |
Function == "K01961" |
Function == "K01962" |
Function == "K01963" |
Function == "K02160" |
Function == "K18472" |
Function == "K11263") 



ps.fatty.acids <- tax_glom(ps.fatty.acids, "Function")



ps_bray <- phyloseq::distance(ps.fatty.acids, method="bray")
ps.ord <- ordinate(ps.fatty.acids, method="PCoA", distance=ps_bray)

plot = plot_ordination(ps.fatty.acids, ps.ord, color = "Treatment")
plot_final = plot + geom_point(size = 2) +
  scale_color_manual(values=palette) +
  theme_classic()

ggsave("PCoA.fattyacids.jpeg", width=6, height=4, units = "in", bg = "transparent")
plot_final
dev.off()
```


#### Upset

```{r}
ps.merged <- tax_glom(ps, "Function")

ps.merged1 <- merge_samples(ps.merged, "Treatment")



ps.merged_presSS.ISse = transform_sample_counts(ps.merged1, function(OTU) ifelse(OTU > 0, 1, OTU))

ps.merged_presSS.ISse <- ps.merged_presSS.ISse %>%
  filter_taxa(function(x) sum(x) > 0, TRUE)


ps.merged_presSS.ISse.df <- data.frame(t(otu_table(ps.merged_presSS.ISse)))

ps.merged_presSS.ISse.df <- rownames_to_column(ps.merged_presSS.ISse.df)
  
colnames(ps.merged_presSS.ISse.df) <- c("value", "A", "B", "C", "D", "E", "F")


p1 <- upset(ps.merged_presSS.ISse.df, sets = c("A", "B", "C", "D", "E", "F"), order.by = "freq")



ggsave("Upsetplot.jpeg", width=10, height=20, units = "in", bg = "transparent")
p1
dev.off()


ggplot2::ggsave(ggplotify::as.ggplot(p1),
                filename = fs::path("Upsetplot.jpeg"),
                width=20, height=5, units = "in", bg = "transparent")
```

